src/erlastic_search.erl

```diff
@@ -648,6 +650,30 @@ aliases(Body) ->
 aliases(Params, Body) ->
     erls_resource:post(Params, filename:join([<<"_aliases">>]), [], [], erls_json:encode(Body), Params#erls_params.http_client_options).
 
+%% Bulk operations with default params
+-spec bulk_operation(list()) -> {ok, list} | {error, any()}.
```

Comment:
Need a more descriptive spec than this. For a library like this, specs also serve as documentation. Here, Your function expects a very precise format of lists: the spec should enumerate all possibilities.